sound/soc/codecs/sdw-mockup.c

```diff
@@ -45,31 +41,15 @@ static const struct snd_soc_component_driver snd_soc_sdw_mockup_component = {
 static int sdw_mockup_set_sdw_stream(struct snd_soc_dai *dai, void *sdw_stream,
 				     int direction)
 {
-	struct sdw_stream_data *stream;
-
-	if (!sdw_stream)
-		return 0;
-
-	stream = kzalloc(sizeof(*stream), GFP_KERNEL);
-	if (!stream)
-		return -ENOMEM;
-
-	stream->sdw_stream = sdw_stream;
-
-	/* Use tx_mask or rx_mask to configure stream tag and set dma_data */
-	snd_soc_dai_dma_data_set(dai, direction, stream);
+	snd_soc_dai_dma_data_set(dai, direction, sdw_stream);
```

Comment:
wow, it did this, really?

Reply:
@ujfalusi This was in the original code when I took over in 2018

````c
struct sdw_stream_data {
    struct sdw_stream_runtime *sdw_stream;
    int num_channels;
};

	if (stream->num_channels)
		num_channels = stream->num_channels;
	else
		num_channels = params_channels(params);

int rt700_channel_map(struct snd_soc_dai *dai,
		unsigned int tx_num, unsigned int *tx_slot,
		unsigned int rx_num, unsigned int *rx_slot)
{
	struct sdw_stream_data *dma;

	if (tx_num) {
		dma = dai->playback_dma_data;
		dma->num_channels = tx_num;
	} else {
		dma = dai->capture_dma_data;
		dma->num_channels = rx_num;
	}
````

but the set_channel_map and num_channels was removed, so the whole thing is useless now.

in hindsight, I also don't recall why we added a set_stream() API which is now limited to set_dmadata() now. Oh well.